src/AaveV2CollectorContractConsolidation.sol

```diff
@@ -0,0 +1,258 @@
+            300,
+            0x2885d15b8Af22648b98B122b22FDF4D2a56c6023,
+            ERC20(AZRX).decimals(),
+            8,
+            false
+        );
+        assets[AENS] = Asset(
+            ERC20(AENS).balanceOf(AaveV2Ethereum.COLLECTOR),
+            300,
+            0x5C00128d4d1c2F4f652C267d7bcdD7aC99C16E16,
+            ERC20(AENS).decimals(),
+            8,
+            false
+        );
+        assets[AUST] = Asset(
+            ERC20(AUST).balanceOf(AaveV2Ethereum.COLLECTOR),
+            200,
+            0xa20623070413d42a5C01Db2c8111640DD7A5A03a,
+            ERC20(AUST).decimals(),
+            18,
+            false
+        );
+
+        USDC_DECIMALS = USDC.decimals();
+        ETH_USD_ORACLE_DECIMALS = AggregatorV3Interface(ETH_USD_FEED).decimals();
+    }
+
+    /// @notice Swaps USDC for specified token
+    /// @param _token the address of the token to swap USDC for
+    /// @param _amountOut the amount of token wanted
+    /// @dev User has to approve USDC transfer prior to calling swap
+    function swap(address _token, uint256 _amountOut) external {
+        if (_amountOut == 0) revert OnlyNonZeroAmount();
+
+        uint256 amountIn = getAmountIn(_token, _amountOut);
+        uint256 quantity = assets[_token].quantity;
+        uint256 sendAmount = _amountOut == type(uint256).max ? quantity : _amountOut;
+
+        assets[_token].quantity = quantity - sendAmount;
+
+        USDC.transferFrom(msg.sender, AaveV2Ethereum.COLLECTOR, amountIn);
+        ERC20(_token).safeTransferFrom(AaveV2Ethereum.COLLECTOR, msg.sender, sendAmount);
+        emit Swap(address(USDC), _token, amountIn, sendAmount);
+    }
+
+    /// @notice Returns amount of USDC to be spent to swap for token
+    /// @param _token the address of the token to swap
+    /// @param _amountOut the amount of token wanted
+    /// return amountInWithDiscount the amount of USDC used minus premium incentive
+    /// @dev User check this function before calling swap() to see the amount of USDC required
+    function getAmountIn(address _token, uint256 _amountOut) public view returns (uint256 amountIn) {
+        Asset memory asset = assets[_token];
+        if (asset.oracle == address(0)) revert UnsupportedToken();
+
+        if (_amountOut == type(uint256).max) {
+            _amountOut = asset.quantity;
+        } else if (_amountOut > asset.quantity) {
+            revert NotEnoughTokens();
+        }
+
+        uint256 oraclePrice = getOraclePrice(asset.oracle);
+        uint256 exponent = asset.decimals + asset.oracleDecimals - USDC_DECIMALS;
+
+        if (asset.ethFeedOnly) {
+            uint256 ethUsdPrice = getOraclePrice(ETH_USD_FEED);
+            oraclePrice *= ethUsdPrice;
+            exponent += ETH_USD_ORACLE_DECIMALS;
+        }
+
+        // Basis points arbitrage incentive
+        amountIn =
+            (((_amountOut * oraclePrice) / 10**exponent) * 10000) / // Amount before discount
+            (10000 + asset.premium);
```

Comment:
Would recommend doing all the multiplications first followed by all the divisions so that no precision is lost here. I would also highly recommend adding in comments before this statement to show the calculation exactly so that it's clear to anyone looking at the code to know what exactly is going on here.